onnx/version_converter/BaseConverter.h

```diff
@@ -0,0 +1,78 @@
+// Version converter interface for ONNX models between different opset versions.
+
+#pragma once
+
+#include "onnx/common/ir.h"
+#include "onnx/common/ir_pb_converter.h"
+#include "onnx/common/stl_backports.h"
+#include "onnx/proto_utils.h"
+#include "onnx/defs/schema.h"
+#include <utility>
+#include <iostream>
+#include <stdlib.h>
+// TODO: Remove this import once actual adapters are imported
+#include "onnx/version_converter/adapters/adapter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+// TODO: Consider creating interface for this class.
+class BaseVersionConverter {
+  // Schema for adapters: {<op_name>:{<from_domain>$<from_version>:{<to_domain>
+  // <to_version>: adapter}}}
+  protected:
+    std::unordered_map<std::string, std::unordered_map<std::string, std::unordered_map<std::string, std::unique_ptr<Adapter>>>> adapters;
+
+  public:
+    BaseVersionConverter() = default;
+
+    virtual ~BaseVersionConverter() = default;
+
+    // adapter_lookup should be called in convert_version when the user would
+    // like to identify the proper registered adapter in the adapters map for
+    // a given Node from a certain version to another. It should only be called
+    // when the user knows that an adapter should exist for the given context.
+    const Adapter& adapter_lookup(const Node* op,
+        const OpSetID& initial_version,
+        const OpSetID& target_version) const {
+      const std::string& op_name = op->kind().toString();
```

Comment:
`.toString()` likely returns a temporary object, not a reference to some object stored inside `op`. So if you just keep a reference (i.e. pointer) to it, the object itself will not live anymore when you try to access it later through the reference. You have to use non-references here.